bndtools.core/src/bndtools/views/repository/RepositoriesView.java

```diff
@@ -303,6 +303,16 @@ public void doubleClick(DoubleClickEvent event) {
                     Object element = selection.getFirstElement();
                     if (element instanceof IAdaptable) {
                         URI uri = (URI) ((IAdaptable) element).getAdapter(URI.class);
+                        if (uri == null && element instanceof RepositoryEntry) {
+                            boolean download = MessageDialog.openQuestion(getSite().getShell(), "Repositories", "This repository entry is unable to be opened because it has not been downloaded. " + "Download and open it now?");
+                            if (download) {
```

Comment:
Should the download occur here on the UI thread? Perhaps it should be pushed into a background job?